tests/integration/test_spec.py

```diff
@@ -43,25 +43,146 @@ def test_add_patch(specfile, patch_id_digits):
         """
     )
     assert patch_lines in specfile.read_text()
+
+
+simple_patch = """\
+# A bright yellow patch
+Patch1: yellow.patch
+"""
+
+patch_with_url = """\
+# A bright yellow patch
+Patch0001  :  https://inter.netz/yellow.patch\n
+"""
+
+multiple_patches = """\
+# An interesting purple
+Patch0200: purple.patch
+
+# Orange is a fruit
+# And orange is a colour
+Patch0301: orange.patch
+"""
+
+hanging_comments = """\
+# This comment doesn't belong to any patch
+
+Patch: surprise.patch
+"""
+
+no_space = """\
+# An interesting purple
+Patch0200: purple.patch
+# Orange is a fruit
+Patch0301: orange.patch
+"""
+
+
+@pytest.mark.parametrize(
+    "lines,files,expectation",
+    [
+        (simple_patch, ["yellow.patch"], {"yellow.patch": ["A bright yellow patch"]}),
+        (patch_with_url, ["yellow.patch"], {"yellow.patch": ["A bright yellow patch"]}),
+        (
+            multiple_patches,
+            ["purple.patch", "orange.patch"],
+            {
+                "purple.patch": ["An interesting purple"],
+                "orange.patch": ["Orange is a fruit", "And orange is a colour"],
+            },
+        ),
+        (hanging_comments, ["surprise.patch"], {"surprise.patch": []}),
+        (
+            no_space,
+            ["purple.patch", "orange.patch"],
+            {
+                "purple.patch": ["An interesting purple"],
+                "orange.patch": ["Orange is a fruit"],
+            },
+        ),
+    ],
+    ids=[
+        "simple-patch",
+        "patch-with-url",
+        "multiple-patches",
+        "hanging-comments",
+        "no-space",
+    ],
+)
+def test_read_patch_comments(specfile, lines, files, expectation):
+    """Check reading comment lines that belong to patches"""
+    content = specfile.read_text()
+    content = content.replace("### patches ###\n", lines)
+    specfile.write_text(content)
+    for patch_file in files:
+        Path(specfile.parent, patch_file).touch()
+    spec = Specfile(specfile, sources_dir=specfile.parent)
+    comments = spec.read_patch_comments()
+    assert comments == expectation
+
+
+@pytest.mark.parametrize(
+    "lines,digits",
+    [
+        ("Patch0001 : some.patch\n", 4),
+        ("Patch003000 : some.patch\n", 6),
+        ("Patch: some.patch\n", 0),
+        ("Patch21: some.patch\n", 1),
+        ("Patch9: some.patch\n", 1),
+    ],
+)
+def test_patch_id_digits(specfile, lines, digits):
+    """Check detecting the number of digits used for patch IDs (indices)"""
+    content = specfile.read_text()
+    content = content.replace("### patches ###\n", lines)
+    specfile.write_text(content)
+    spec = Specfile(specfile, sources_dir=specfile.parent)
+    assert spec.patch_id_digits == digits
+
+
+def test_remove_patches(specfile):
+    """Check patches being removed from a specfile"""
+    no_patches = specfile.read_text().replace("### patches ###\n", "")
+    patches = no_patches.replace(
+        "### patches ###\n",
```

Comment:
I think this is no-op since `no_patches` has "### patches ###\n" already removed/replaced

```suggestion
    patches = specfile.read_text().replace(
        "### patches ###\n",
```

Reply:
Thanks! Fixed.